unity-renderer/Assets/Scripts/MainScripts/DCL/Components/TextShape/TextShape.cs

```diff
@@ -61,50 +61,57 @@ public class Model : BaseModel
         private Model cachedModel;
         private Material cachedFontMaterial;
         private Camera mainCamera;
+        private bool cameraFound;
 
         public override string componentName => "text";
 
         private void Awake()
         {
             model = new Model();
-            mainCamera = Camera.main;
 
             cachedFontMaterial = new Material(text.fontSharedMaterial);
             text.fontSharedMaterial = cachedFontMaterial;
             text.text = string.Empty;
-            
+
             Environment.i.platform.updateEventHandler.AddListener(IUpdateEventHandler.EventType.Update, OnUpdate);
         }
 
         private void OnUpdate()
         {
-            if (cachedModel.billboard && mainCamera != null)
+            // Cameras are not detected while loading, so we can not load the camera on Awake or Start
+            if (!cameraFound)
             {
-                transform.forward = mainCamera.transform.forward;
+                mainCamera = Camera.main;
+                if (mainCamera != null)
+                    cameraFound = true;
             }
+
+            if (cachedModel.billboard && mainCamera != null)
```

Comment:
You have added here the `cameraFound` but in the line bellow we anyway check for `mainCamera`. Could we combine the optimization?

Reply:
Agree, I think we can add that, if there is a teleport removing the camera, the billboard is also removed, so it should be no problem at all